lib/core/constants.js

```diff
@@ -102,10 +102,8 @@ const wellknownHeaderNames = /** @type {const} */ ([
 ])
 
 /** @type {Record<typeof wellknownHeaderNames[number]|Lowercase<typeof wellknownHeaderNames[number]>, string>} */
-const headerNameLowerCasedRecord = {}
-
 // Note: object prototypes should not be able to be referenced. e.g. `Object#hasOwnProperty`.
-Object.setPrototypeOf(headerNameLowerCasedRecord, null)
+const headerNameLowerCasedRecord = Object.create(null);
```

Comment:
Currently, I only replace with `Object.create(null)`.

Reply:
I noticed that there is also https://github.com/nodejs/undici/blob/d6c44f3b49949dbdd66d9c56e729d503e6b09ddc/lib/core/util.js#L822-L845 and seems to be a bit redundant. Should I change the initialization of that with `Object.create(null)` as well?

cc @tsctx @mcollina

Reply:
I think you can use `__proto__: null`.

Reply:
Done in [`f736e90` (#3726)](https://github.com/nodejs/undici/pull/3726/commits/f736e908934c5c56e443ce9872d60e7351d93875)